CHANGES.md

```diff
@@ -1,5 +1,14 @@
 # Change Log
 
+## v0.7.8
```

Comment:
Shouldn't version number changes be handled separately from functional pull requests, as an accumulation of all the things that got rolled up since the last release?

Reply:
Part of the deployment process [automatically creates a tag based upon the pack version](https://github.com/StackStorm-Exchange/ci/blob/master/.circle/deployment#L86). That way people can install a specific pack version using `st2 pack install <pack>=<version>`.

Users that don't specify a version when installing a pack will get the `master` branch.

If we merge changes to master without bumping the pack version, it gets difficult to know what code a user has, since the version in their `pack.yaml` might not match any particular git tag.

This overall process could be changed, but it requires working through multiple pieces around the different pack repos, and the pack management code.   

We also tend not to have larger pack releases combining code from multiple PRs. It instead tends to be individual changes. Waiting for a larger group of changes to do a bulk release wouldn't work all that well for most packs, which do not see a high rate of change.

Reply:
By your logic every commit to master needs a version bump, that sounds excessive and unnecessary.  Most projects accumulate changes and then make a release.  I expect this pack to see a high rate of changes over the next year.

Reply:
You are welcome to propose an alternative method, but it needs to take into account the overall pack management dynamics. We’re not just going to start making in-versioned changes in this one pack. 

You would have to look at the overall pack management workflow, and come up with a comprehensive update, not just change things here because it feels unecessary to you.

Reply:
Projects typically accumulate changes and then roll a release.  I don't see why this would be different, however I would guess the root cause is that "st2 pack install" defaults to master.  This means folks installing packs are always taking things from master which means master is always a stable and versioned entity?  This probably derives from packs having a small number of participants in them until now.  That won't scale into projects that have many participants.  If you end up with 10 PRs coming in each day into master, all of them have to submit pack.yaml updates, so you have to merge one at a time and then force the 9 others to rebase to fix the pack.yaml collision, then merge another and force the remaining 8 to rebase to fix the pack.yaml collision, etc...

The boost C++ project manages this by having all PRs for a repo (see: pack) flow into a "develop" branch, and the maintainer of each library (see: pack) will merge develop into "master" to declare that all accumulated changes are ready for release.  This would work well for pack development and ensure scalability.  If 10 people submit PRs they all go into the "develop" branch, and then at some point a release manager merge (fast-forward) develop into master, including changes to pack.yaml version (atomic, they all land together), and then a release is tagged from that.  This would allow PRs to flow in without conflict, and would allow the project to accumulate multiple changes in a single public update.

Reply:
@jeking3 That is indeed how it works on the main StackStorm/st2 repo.

However, the activity on Exchange packs is just not high enough to warrant a complex versioning and release strategy.

I think it would be healthy to explore this further though and maybe brainstorm some ideas, so that when pack's do receive more contributions we will be ready. Would you mind creating an issue on the StackStorm/st2 repo about "How can we improve the overall pack development and release process"?

 Thank you for your contributions and insight!

Reply:
Sure, will do.  Thanks.